WalletWasabi.Fluent/ViewModels/Wallets/Send/FeeChartViewModel.cs

```diff
@@ -255,7 +256,17 @@ public void UpdateFeeEstimates(Dictionary<int, int> feeEstimates, FeeRate? maxFe
 		SliderMaximum = confirmationTargetValues.Length - 1;
 		var confirmationTargetCandidate = ConfirmationTargetValues.MinBy(x => Math.Abs(x - Services.UiConfig.FeeTarget));
 		CurrentConfirmationTarget = Math.Clamp(confirmationTargetCandidate, ConfirmationTargetValues.Min(), ConfirmationTargetValues.Max());
-		SliderValue = GetSliderValue(CurrentConfirmationTarget, ConfirmationTargetValues);
+
+		if (!_hasPreviousSliderValue)
+		{
+			_hasPreviousSliderValue = true;
+			SliderValue = GetSliderValue(CurrentConfirmationTarget, ConfirmationTargetValues);
+		}
+		else
+		{
+			SetXAxisCurrentValue(SliderValue, true);
+		}
```

Comment:
I think this is buggy now.

If the current values are `1,2,3,4,5` and the selected index is `4`.
Now if a new estimation received `1,2`, it will crash in `SetXAxisCurrentValue` because the SliderValue is not updated. And the index will be out of range.

I think what you need to do is simply set a default value for the `SliderValue`

```c#
[AutoNotify] private int _sliderValue = -1;
```

And modify the line that sets the value of `confirmationTargetCandidate`:
```c#
var confirmationTargetCandidate = SliderValue == -1
			? ConfirmationTargetValues.MinBy(x => Math.Abs(x - Services.UiConfig.FeeTarget))
			: SliderValue;
```

So the next line will ensure that the index will be always in the range.
https://github.com/zkSNACKs/WalletWasabi/blob/3e151ca170480dbf2de8e652c1316b3bfd8bf532/WalletWasabi.Fluent/ViewModels/Wallets/Send/FeeChartViewModel.cs#L257

What do you think?

Reply:
That is perfect! I'm testing your solution right now. Let me check if it works as expected.